src/longestCommonPxth.ts

```diff
@@ -0,0 +1,34 @@
+import { createPxth } from './createPxth';
+import { getPxthSegments } from './getPxthSegments';
+import { Pxth } from './Pxth';
+import { PxthSegments } from './PxthSource';
+
+/**
+ * Finds longest common path.
+ * @example
+ * [['hello', 'world'], ['hello', 'world, 'yes'], ['hello', 'world', 'bye', 'asdf']] -> 'hello.world'
+ * [['a'], ['b'], ['c']] -> RootPathToken
+ */
+export const longestCommonPxth = (paths: Pxth<unknown>[]): Pxth<unknown> => {

```

Comment:
I don't know if it is possible to make this function faster. O(N^2) in worst case now.

Reply:
The asymptotic complexity of this algorithm is O(N*K), where K is the length of the first path and N is the total amount of paths, passed into function arguments. And yes, I think there is no better solution without overengineering this problem.